test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp

```diff
@@ -0,0 +1,148 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/type_traits>
+#include <oneapi/dpl/cstddef>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+#include "support/utils_invoke.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <class, class = dpl::void_t<>>
+struct has_type_member : dpl::false_type
+{
+};
+
+template <class T>
+struct has_type_member<T, dpl::void_t<typename dpl::underlying_type<T>::type>> : dpl::true_type
+{
+};
```

Comment:
We defined the same structure in `result_of.pass.cpp`. I propose to move it into common utils header.

Reply:
During this PR the code `struct has_type_member` exist only in this file.
So I propose to fix this later.

Reply:
No, as I see it exists twice. The first one is here and the second one is `HasType` structure in `result_of.pass.cpp` test. 
They are semantically the same. In terms of this PR, `HasType<dpl::underlying_type<T>> can be used. 
I just propose not to introduce duplicated functionality as part of this PR.

Reply:
Fixed